ProcessesApi/V1/Services/ChangeOfNameService.cs

```diff
@@ -76,6 +77,25 @@ private async Task ReviewDocumentsCheck(StateMachine<string, string>.Transition
             await TriggerStateMachine(processRequest).ConfigureAwait(false);
         }
 
+        public async Task CheckTenureInvestigation(StateMachine<string, string>.Transition transition)
+        {
+            var processRequest = transition.Parameters[0] as ProcessTrigger;
+
+
+            var triggerMappings = new Dictionary<string, string>
+            {
+                { SharedValues.Appointment, SharedInternalTriggers.TenureInvestigationPassedWithInt },
+                { SharedValues.Approve, SharedInternalTriggers.TenureInvestigationPassed },
+                { SharedValues.Decline, SharedInternalTriggers.TenureInvestigationFailed }
+            };
+
+            processRequest.HandleRecommendation(triggerMappings,
```

Comment:
Finally thought of a good name for this method! How about `SelectTriggerFromUserInput` or select from form data?

Also does this method have unit tests?